miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/RealMiniApp.kt

```diff
@@ -30,6 +30,22 @@ internal class RealMiniApp(
     override fun setCustomPermissions(miniAppCustomPermission: MiniAppCustomPermission) =
         miniAppCustomPermissionCache.storePermissions(miniAppCustomPermission)
 
+    @Suppress("FunctionMaxLength")
+    override fun listDownloadedWithCustomPermissions(): List<MiniAppInfo> {
+        val result: ArrayList<MiniAppInfo> = arrayListOf()
+        val unfilteredMiniApps = miniAppDownloader.getDownloadedMiniAppList()
+        val filteredMiniApps =
+            miniAppCustomPermissionCache.readAllStoredPermissions(unfilteredMiniApps)
+
+        filteredMiniApps.forEach { (miniAppId) ->
```

Comment:
This should return all Mini Apps which have been downloaded, whether they have any permissions granted or not. So no need to filter here. 

Mini Apps which are downloaded but don't have any permissions granted should just have all permissions set to DENIED.

Reply:
Thanks, I'll update this part. Also, I'll keep the same name as `listDownloadedWithCustomPermissions` for the public interface.